### PR TITLE
Add BoundedVec to Treasury Pallet

### PR DESCRIPTION
**Pull request by shawntabrizi**
Helps with https://github.com/paritytech/substrate/issues/8629

This adds an upper bound to the number of treasury proposals that can be in the queue for spending.

This will be used for benchmarking the PoV size.

polkadot companion: https://github.com/paritytech/polkadot/pull/2932

### CODE REVIEW

File: frame/treasury/src/lib.rs [hunk 1]
```diff
@@ -64,10 +64,13 @@ mod benchmarking;
 pub mod weights;
 
 use sp_std::prelude::*;
-use frame_support::{decl_module, decl_storage, decl_event, ensure, print, decl_error, PalletId};
+use frame_support::{
+	decl_module, decl_storage, decl_event, ensure, print, decl_error,
+	PalletId, BoundedVec, bounded_vec::TryAppendValue,
```
Comment (kianenigma) [inline frame/treasury/src/lib.rs:69]:
```suggestion
	PalletId, BoundedVec,
```

Hmm are you sure we need this? We implemented it directly for `storage::types` so we won't need to import again.

**Review by kianenigma — APPROVED** [review]

**Review by coriolinus — APPROVED** [review]

File: frame/treasury/src/lib.rs (hunk 1 above)
Comment (shawntabrizi) [reply frame/treasury/src/lib.rs:69]: did get a compiler issue. let me see

Comment (shawntabrizi) [reply frame/treasury/src/lib.rs:69]:
```
error[E0599]: no function or associated item named `try_append` found for struct `Approvals<T, I>` in the current scope
   --> frame/treasury/src/lib.rs:329:23
    |
178 | decl_storage! {
    | ------------- function or associated item `try_append` not found for this
...
329 |             Approvals::<T, I>::try_append(proposal_id).map_err(|_| Error::<T, I>::TooManyApprovals)?;
    |                                ^^^^^^^^^^ function or associated item not found in `Approvals<T, I>`
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
66  | use crate::sp_api_hidden_includes_decl_storage::hidden_include::bounded_vec::TryAppendValue;
```

If you know a better fix, happy to accept a follow up pr

**Comment by shawntabrizi** [comment]
bot merge

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**